lala-referral-web/Makefile

```diff
@@ -0,0 +1,35 @@
+.PHONY: setup build run debug
+CYAN:=\033[0;36m
+RED=\033[0;31m
+NC:=\033[0m
+APP:=lala.tg.bot
+
+.SILENT:
+setup:
+	echo -e "${CYAN}Installing dependencies${NC}"
+	npm install --silent \
+	&& echo -e "${CYAN}Setup finished run make build to proceed!${NC}" \
+	|| echo -e "${RED}Setup Failed!${NC}"
+
+build:
+	echo -e "${CYAN}Building environment:${NC}"
+	docker build --rm -t ${APP} . \
+	&& echo -e "${CYAN}Application is ready to run!${NC}" \
+	|| echo -e "${RED}Build Failed!${NC}"
+
+run:
+	echo -e "${CYAN}Starting up application${NC}"
+	docker run -it --rm -v ${PWD}/:/app/ -p 3000:3000 ${APP}
+
+test:
+	echo -e "${CYAN}Installing dependencies${NC}"
+	docker run -it --rm -v ${PWD}/:/app/ ${APP} npm test \
+	&& echo -e "${CYAN}Test Finished${NC}" 
```

Comment:
so you are running a new container here, which has no node package installed, I bet this build target will always fail.

Reply:
If you use docker compose you can write it very simply like
```
up:
    docker-compose up
test: up
    docker-compose exec -it your-service-ame npm test
```